### PR TITLE
raftstore: accelerate ticks for old region leader after split

### PR DESCRIPTION
**Pull request by hhkbp2**
This PR tries to fix issue #1538

### CODE REVIEW

**Comment by hhkbp2** [comment]
PTAL @siddontang @BusyJay

**Comment by BusyJay** [comment]
LGTM

**Comment by siddontang** [comment]
If user pass another election timeout from config, we must always reset the accelerate ticks too. I think we can set the tick to election timeout  - 1 in config validate function.

so we can use election timeout - 1 directly, no need to declare another variable.

File: src/raftstore/store/config.rs [hunk 1]
```diff
@@ -158,8 +157,7 @@ impl Default for Config {
             lock_cf_compact_interval: DEFAULT_LOCK_CF_COMPACT_INTERVAL,
             consistency_check_tick_interval: DEFAULT_CONSISTENCY_CHECK_INTERVAL,
             report_region_flow_interval: DEFAULT_REPORT_REGION_FLOW_INTERVAL,
-            leader_accelerate_campaign_after_split_ticks:
-                DEFAULT_LEADER_ACCELERATE_CAMPAIGN_AFTER_SPLIT_TICKS,
+            leader_accelerate_campaign_after_split_ticks: RAFT_ELECTION_TIMEOUT_TICKS - 1,
```
Comment (siddontang) [inline src/raftstore/store/config.rs]: I prefer naming accelerate_campaign_after_split_ticks, here it is not leader.

**Comment by siddontang** [comment]
PTAL @BusyJay

File: src/raftstore/store/config.rs [hunk 2]
```diff
@@ -117,6 +117,12 @@ pub struct Config {
     pub consistency_check_tick_interval: u64,
 
     pub report_region_flow_interval: u64,
+
+    /// For the peer which is the leader of the region before split,
+    /// `leader_accelerate_campaign_after_split_ticks` specifies the tick number to be accelerated
+    /// after the region split, so that the peer of new split region may campaign and become leader
+    /// earlier than other follower peers.
+    pub leader_accelerate_campaign_after_split_ticks: usize,
```
Comment (BusyJay) [inline src/raftstore/store/config.rs]: We should provide a function instead.

**Comment by BusyJay** [comment]
LGTM

**Comment by siddontang** [comment]
LGTM

**Pull request merged**